src/app/bundles/document-selection/document-selection.component.ts

```diff
@@ -8,11 +9,38 @@ import { MockCaseDocuments } from '../shared/mockcasedocumentdata';
 })
 export class DocumentSelectionComponent implements OnInit {
 
-  cases = MockCaseDocuments;
+  documents = MockCaseDocuments;
+  selectAllStatus: boolean;
+
+  @ViewChildren('documentItem') documentItem: QueryList<DocumentItemComponent>;
 
   constructor() { }
 
   ngOnInit() {
+    this.selectAllStatus = false;
+  }
+
+  updateSelectAll() {
+    this.selectAllStatus = !this.selectAllStatus;
+    this.documentItem.map((document: DocumentItemComponent) => document.checked = this.selectAllStatus);
```

Comment:
map is used to map one to another. If you are returning the return type value of map it means you probably want a for loop. You can use `for of` or if you want a one liner `.forEach(...)`

Reply:
you are right, i think for best practice, I changed to forEach